### DISCUSSION TITLE
Is it possible to disable control flow statements?

### DISCUSSION THREAD

**Discussion by joaooliveirapro** [handlebars-lang/handlebars.js#1968] [Q&A] [answered]
Hi

Is there a way to disable access to control flow (if, else, unless, with, etc.)?

For my use case, I need a templating engine that provides efficient keyword replacement ({{ keyword }}) but I don't want users to be able to use control flow statements. 

Essentially, if the control flow is disabled, then when evaluating the template the engine would ignore the control flow statements. 

```
const userTemplate = '<h1>Title {{ title }}</h1>{{#if condition}} show {{/if}}';  
const template = Handlebars.compile(userTemplate);
const context = { title: 'My App', condition: true};

const result = template(context);  // currently: '<h1>Title My App</h1>show'
const result = template(context);  // expected with control flow disable:  '<h1>Title My App</h1>{{#if condition}} show {{/if}}' // gets ignored. 
```

I couldn't find this on the docs. 

Thanks
Joao

**Comment by jaylinski** [comment, marked as answer]
You can use `Handlebars.unregisterHelper()` for [built-in helpers](https://handlebarsjs.com/guide/builtin-helpers.html):

https://handlebarsjs.com/playground.html#format=1&currentExample=%7B%22template%22%3A%22%7B%7B%23if%20author%7D%7D%5CnAuthor%3A%20%7B%7BfirstName%7D%7D%20%7B%7BlastName%7D%7D%5Cn%7B%7B%2Fif%7D%7D%22%2C%22partials%22%3A%5B%5D%2C%22input%22%3A%22%7B%5Cn%20%20author%3A%20true%2C%5Cn%20%20firstName%3A%20%5C%22Yehuda%5C%22%2C%5Cn%20%20lastName%3A%20%5C%22Katz%5C%22%2C%5Cn%7D%5Cn%22%2C%22output%22%3A%22Author%3A%20Yehuda%20Katz%5Cn%22%2C%22preparationScript%22%3A%22Handlebars.unregisterHelper('if')%3B%5Cn%22%2C%22handlebarsVersion%22%3A%224.7.7%22%7D

This will make compilation fail if an unregistered helper is used.

Or you can just overwrite existing helpers like this:

https://handlebarsjs.com/playground.html#format=1&currentExample=%7B%22template%22%3A%22%7B%7B%23if%20author%7D%7D%5CnAuthor%3A%20%7B%7BfirstName%7D%7D%20%7B%7BlastName%7D%7D%5Cn%7B%7B%2Fif%7D%7D%22%2C%22partials%22%3A%5B%5D%2C%22input%22%3A%22%7B%5Cn%20%20author%3A%20true%2C%5Cn%20%20firstName%3A%20%5C%22Yehuda%5C%22%2C%5Cn%20%20lastName%3A%20%5C%22Katz%5C%22%2C%5Cn%7D%5Cn%22%2C%22output%22%3A%22Author%3A%20Yehuda%20Katz%5Cn%22%2C%22preparationScript%22%3A%22Handlebars.registerHelper('if'%2C%20function%20(conditional%2C%20options)%20%7B%5Cn%20%20%20%20return%20options.fn(this)%3B%5Cn%7D)%3B%5Cn%22%2C%22handlebarsVersion%22%3A%224.7.7%22%7D

**Comment by joaooliveirapro** [comment]
Thank you! This is great
